src/cljc/spy/test.cljc

```diff
@@ -96,46 +104,59 @@
                         ~`expected-not-called
                         ~`actual-not-called))
 
+     (defmacro assert-call-matching
+       [msg form report-fn]
+       `(assert-called ~msg
+                       ~form
+                       ~report-fn
+                       (fn [~'& ~'args]
+                         (expected-call-matching '~(last form)))
+                       ~`actual-call-matching))
+
      (defmethod t/assert-expr 'spy/called-n-times?
        [msg form]
-       `(assert-called-n-times ~msg ~form t/do-report))
+       `(assert-called-n-times ~msg ~form ~`t/do-report))
```

Comment:
Using ``~`t`` here to ensure that `t` is fully expanded to `clojure.test` (and `cljs.test` in the CLJS implementation).